swarm/src/lib.rs

```diff
@@ -1677,6 +1683,68 @@ impl error::Error for DialError {
     }
 }
 
+/// Possible errors when upgrading an inbound connection.
+#[derive(Debug)]
+pub enum ListenError {
```

Comment:
Mentioning it here, though not directly related: I always confuse `LIstenError`, signaling a failure on upgrading an inbound connection, with the failure of listening on a new address. Anyone else?

Reply:
A lot of naming could be improved here :)

I believe this is consistent with the naming of `DialError` but I agree that something like `InboundConnectionError` and `OutboundConnectionError` would probably be better.

I'll see to tidy up these when we remove the connection limits and banned variants.